platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java

```diff
@@ -157,7 +158,9 @@ public Connection createConnection(final NodeId otherId) {
      * @return the IP address to connect to
      */
     private byte[] getConnectAddressIpv4(final Address from, final Address to) {
-        if (from.isOwnHost() && to.isOwnHost() && settings.useLoopbackIp()) {
+        final boolean fromIsLocal = AddressBookNetworkUtils.isLocal(from);
```

Comment:
Why is `isLocal` not a method of `Address`?

Reply:
Because `Address` and `AddressBook` are serialized as part of the state and the association between an address and an executing node is logic outside the address/addressbook itself.    Everyone gets the same state and addressbook.  But not everyone is going to have the same result for "isLocal".   Divergence in a synchronized data structure is not good design.